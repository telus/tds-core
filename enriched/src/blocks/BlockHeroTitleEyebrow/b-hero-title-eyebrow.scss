@import '~telus-thorium-core/scss/settings/variables';

.b-hero-title-eyebrow {
  &:after {
    padding-bottom: 8.13037%;
    background-image: url(//cdn.business.telus.digital/v2-assets/images/b4yb/waves/wave_landing.png);
    background-position: center bottom;
    background-repeat: no-repeat;
    background-size: 100%, auto, contain;
    bottom: -1px;
    content: '';
    left: 0;
    position: absolute;
    right: 0;
  }

  &__eyebrow {
    color: $color-grape;
  }
}
