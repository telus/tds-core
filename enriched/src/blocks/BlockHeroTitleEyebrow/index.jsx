import React from 'react';

if ( process.env.BROWSER ) {
  require('./b-hero-title-eyebrow.scss');
}

const BlockHeroTitleEyebrow = (props) => {
  const { eyebrow, title, heroImage } = props;

  const headerImage = {
    backgroundImage: `url(${heroImage.file.url})`,
    paddingBottom: `28.125%`,
    backgroundPosition: `center top`,
    backgroundSize: `cover`,
    position: `relative`,
    width: `100%`
  };

  return (
    <section>
      <div className="b-hero-title-eyebrow" style={headerImage} />
      <div className="container container--limited-width">
        <div className="grid-row">
          <div className="medium-12">
            <p className="heading-4 b-hero-title-eyebrow__eyebrow">{eyebrow}</p>
            <h1 className="heading-1">{title}</h1>
          </div>
        </div>
      </div>
    </section>
  );
};

BlockHeroTitleEyebrow.propTypes = {
  title: React.PropTypes.string.isRequired,
  eyebrow: React.PropTypes.string,
  heroImage: React.PropTypes.object.isRequired
};

BlockHeroTitleEyebrow.defaultProps = {
  eyebrow: ''
};

export default BlockHeroTitleEyebrow;
