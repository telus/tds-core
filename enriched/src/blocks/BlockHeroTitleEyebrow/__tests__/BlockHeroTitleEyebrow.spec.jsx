import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import BlockHeroTitleEyebrow from '../';

describe('BlockHeroTitleEyebrow', () => {
  const stubHeroImage = { file: { url: 'some/url', title: 'Client logo title' } };
  const data = { eyebrow: 'Foo', title: 'Bar', heroImage: stubHeroImage };

  it('matches snapshot', () => {
    const wrapper = mount(<BlockHeroTitleEyebrow {...data} />);
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('should render eyebrow', () => {
    const wrapper = mount(<BlockHeroTitleEyebrow {...data} />);
    expect(wrapper.find('p').first().render().text()).toEqual('Foo');
  });

  it('should render without subcategory', () => {
    const wrapper = mount(<BlockHeroTitleEyebrow title={data.title} heroImage={stubHeroImage} />);
    expect(wrapper.find('p').first().render().text()).toEqual('');
  });

  it('should render title', () => {
    const wrapper = mount(<BlockHeroTitleEyebrow {...data} />);
    expect(wrapper.find('h1').first().render().text()).toEqual('Bar');
  });

  it('should use headline style', () => {
    const wrapper = mount(<BlockHeroTitleEyebrow {...data} />);
    expect(wrapper.find('section').children().first().hasClass('b-hero-title-eyebrow')).toBeTruthy();
  });

  it('should add the heroimage as a background', () => {
    const wrapper = mount(<BlockHeroTitleEyebrow {...data} />);
    expect(wrapper.find('.b-hero-title-eyebrow').prop('style')).toEqual({ backgroundImage: 'url(some/url)', backgroundPosition: 'center top', backgroundSize: 'cover', paddingBottom: '28.125%', position: 'relative', width: '100%' });
  });
});
