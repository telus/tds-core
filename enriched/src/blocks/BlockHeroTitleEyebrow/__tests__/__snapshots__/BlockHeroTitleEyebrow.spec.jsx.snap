exports[`BlockHeroTitleEyebrow matches snapshot 1`] = `
<BlockHeroTitleEyebrow
  eyebrow="Foo"
  heroImage={
    Object {
      "file": Object {
        "title": "Client logo title",
        "url": "some/url",
      },
    }
  }
  title="Bar">
  <section>
    <div
      className="b-hero-title-eyebrow"
      style={
        Object {
          "backgroundImage": "url(some/url)",
          "backgroundPosition": "center top",
          "backgroundSize": "cover",
          "paddingBottom": "28.125%",
          "position": "relative",
          "width": "100%",
        }
      } />
    <div
      className="container container--limited-width">
      <div
        className="grid-row">
        <div
          className="medium-12">
          <p
            className="heading-4 b-hero-title-eyebrow__eyebrow">
            Foo
          </p>
          <h1
            className="heading-1">
            Bar
          </h1>
        </div>
      </div>
    </div>
  </section>
</BlockHeroTitleEyebrow>
`;
